### PR TITLE
The Great Online Odyssey

### PR DESCRIPTION
**Pull request by Malacath-92**
Inspired by #235 I took another stab at putting a lock on every script access. This should make scripts work online more reliably. Obviously there are still issues, e.g. accessing different state, being able to conjure up race conditions through Lua, etc. So even if this is better it still requires some good thought from a script creator.
Tested with a level gen mod kindly provided by Estebanfer, not tested with anything else yet. But they reported that they were playing with it for half an hour without crashes, while without this PR getting random entity corruption after as early as one level and sooner or later a crash.

### CODE REVIEW

**Comment by Dregu** [comment]
Nothing works. PL doesn't build, OL Clang doesn't build, OL MSVC can't init state and crashes.

**Comment by Malacath-92** [comment]
Oh right, I forgot to push my branch of PL.
Should've been more explicit that I didn't test OL yet. And I usually don't build clang locally, no time to do two long builds

**Comment by Malacath-92** [comment]
It's still a draft PR, didn't expect anyone to try it immediately to be honest  😅

**Comment by Dregu** [comment]
You said it's been tested with good results so I expected it to at least build :)

**Comment by Malacath-92** [comment]
I pushed the changes to PL that are needed to build it. Still working on OL.

File: docs/game_data/spel2.lua [hunk 1]
```diff
@@ -1174,12 +1174,14 @@ function get_rva(address_name) end
 function log_print(message) end
 ---Immediately ends the run with the death screen, also calls the save_progress
 ---@return nil
-function call_death_screen() end
----Saves the game to savegame.sav and displays spinning cog in the bottom right corner
----@return nil
+function load_death_screen() end
```
[the remark this answers is not in the document]
Comment (Mr-Auto) [reply docs/game_data/spel2.lua:1177]: oh, it's only the name for the api, that confused me

[the remark this answers is not in the document]
Comment (Dregu) [reply docs/game_data/spel2.lua:1177]: Yeah we already had `load_screen` so adding a `call_..._screen` was weird

**Pull request merged**